src/services/Pokemon.js

```diff
@@ -0,0 +1,30 @@
+export async function fetchPokemon() {
+  const resp = await fetch(`https://pokedex-alchemy.herokuapp.com/api/pokedex`);
+  const data = await resp.json();
+  return (data.results);
+}
+
+export async function fetchPokemonTypes() {
+  const resp = await fetch(`https://pokedex-alchemy.herokuapp.com/api/pokedex/types`);
+  const data = await resp.json();
+  const types = data.map((item) => item.type);
+  return (types);
+}
+
+export async function fetchFilteredPokemon(type, searchPokemon, order) {
+  const params = new URLSearchParams();
+  params.set('perPage', '10');
+  if (type !== 'all') {
+    params.set('type', type);
+  }
+  if (searchPokemon) {
+    params.set('pokemon', searchPokemon);
+  }
+  if (order !== '') {
+    params.set('sort', 'pokemon');
+    params.set('direction', order);
+  }
```

Comment:
Great use cases here to prevent errors in our return data! Way to think through the complex contingencies!